### PR TITLE
Agent image pipeline - icr.io & public publish 

### PR DESCRIPTION
**Pull request by htreu**
* use manifest resource instead of shell scripts
* manual: publish latest(-j9) & <version>(-j9) images + manifest public

### CODE REVIEW

**Review by tkohn — APPROVED** [review]
PR of concourse is really hard with too many changes 
But for me it looks good 👍 

Awesome job to move the Jenkins pipeline to concourse 💪

File: ci/create-tag-file.yml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+params:
+  COMMIT_SHA: ((commit-sha))
+  CLASSIFIER: ((classifier))
+  VERSION: ((version))
+  FLAVOR: ((flavor))
+  RELEASE_TREE: ((release-tree))
+run:
+  path: /bin/bash
+  args:
+  - -exc
```
Comment (mdonkers) [inline ci/create-tag-file.yml:25]: shouldn't this be `-exec` ?

Comment (mdonkers) [reply ci/create-tag-file.yml:25]: (or what are the parameters for?)

File: ci/create-tag-file.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        CLASSIFIER="-${CLASSIFIER}"
+      fi
+
+      if [ "${RELEASE_TREE}x" = 'x' ]; then
+        RELEASE_TREE="prerelease"
+      fi
+
+      # Only tag with commit SHA on pre-release:
+      COMMIT_SHA_TAG=""
+      if [ "${RELEASE_TREE}x" = 'prereleasex' ]; then
```
Comment (mdonkers) [inline ci/create-tag-file.yml:45]: I think here you don't need to add the `x`. Just `"${RELEASE_TREE}" = 'prerelease'` would be more readable.

In the other places the addition of the `x` is only to handle both empty and not-set variables correctly.

File: ci/create-tag-file.yml [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+      latest$CLASSIFIER
+      $VERSION$CLASSIFIER
+      $COMMIT_SHA_TAG
+      EOF
+
+      # Build Slack message:
+      DOCKER_RELEASE=""
+      REDHAT_RELEASE=""
+      ICR_RELEASE=""
+      if [ "${RELEASE_TREE}x" = 'releasex' ] && [ "${FLAVOR}x" = 'dynamicx' ]; then
```
Comment (mdonkers) [inline ci/create-tag-file.yml:60]: Same here on the variable handling (without the `x`)

File: ci/create-tag-file.yml (hunk 1 above)
Comment (htreu) [reply ci/create-tag-file.yml:25]: * -e is for exiting as soon as a command in the script exits with non-zero
* -x is for printing commands & params as they execute
* -c is for reading the script from stdin, which the following yaml list entry provides

thats at leats my understanding...

**Pull request merged**